spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteViewCommands.scala

```diff
@@ -60,6 +62,9 @@ case class RewriteViewCommands(spark: SparkSession) extends Rule[LogicalPlan] wi
         properties = properties,
         allowExisting = allowExisting,
         replace = replace)
+
+    case AlterViewAs(ResolvedV2View(_, _), _, _) =>
+      throw new AnalysisException("ALTER VIEW <viewName> AS is not supported. Use CREATE OR REPLACE VIEW instead")
```

Comment:
Does this need to happen here instead of in `CheckViews` so that Spark doesn't throw a different analysis exception?

Reply:
I threw the error here mainly because when we supported `AlterViewAs`, most of the things had to happen in `RewriteViewCommands` but I've moved this now to `CheckViews`